File: Programmer/Android/empty-app/app/src/main/java/kamakura/yx/ea/ui/MainActivity.java
package kamakura.yx.ea.ui;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import com.bytedance.platform.tracer.HawTracer;
import com.bytedance.platform.tracer.enhance.FF;

import kamakura.yx.ea.MainApplication;
import kamakura.yx.ea.R;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.i(TAG, "onCreate");
        setContentView(R.layout.activity_main);
        getWindow().getDecorView().postDelayed(
                () -> {
                    HawTracer.i("i compute time duration");
                    Log.w(TAG, "Report app time duration: " + (System.currentTimeMillis() - MainApplication.APP_START_TIME));
                    HawTracer.i("o compute time duration");
                },
                2000
        );
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG, "onResume");
        FF ff = new FF();
        ff.b(1024);
    }
}

File: Programmer/Android/trace-id/app/src/main/java/kamakura/yx/android/ti/MainActivity.kt
package kamakura.yx.android.ti

import android.os.Bundle
import android.os.Process
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.Text
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Surface
import androidx.compose.runtime.Composable
import androidx.compose.ui.platform.setContent
import androidx.ui.tooling.preview.Preview
import kamakura.yx.android.ti.ui.TraceidTheme
import java.util.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            TraceidTheme {
                // A surface container using the 'background' color from the theme
                Surface(color = MaterialTheme.colors.background) {
                    Greeting("Android")
                }
            }
        }

        runOnUiThread {
            Log.i("MainActivity", "${UUID.randomUUID().toString()}-${Process.myPid()}")
        }
    }
}

@Composable
fun Greeting(name: String) {
    Text(text = "Hello $name!")
}

@Preview(showBackground = true)
@Composable
fun DefaultPreview() {
    TraceidTheme {
        Greeting("Android")
    }
}